### ISSUE TITLE
Expo/React Native

### ISSUE THREAD

**Issue by addreeh** [mtcute/mtcute#57]
Does this library works with React Native/Expo projects? For example, gramjs dont work because of crypto, but this library use a different dependency

**Comment by teidesu** [comment]
honestly – no idea. it probably does, but there isn't a ready-made package for react native (yet?)

judging by their docs, `WebSocket`s are supported, so for transport `WebSocketTransport` from `@mtcute/web` should work fine
as for crypto – `@mtcute/web` uses a mix of wasm-based crypto and webcrypto apis, neither of which are supported by react native...
for storage mtcute primarily uses sqlite, and react native does seem to have some libraries that provide bindings for those, but i don't know how well would they work with mtcute

tldr: out of the box – not right now, but i will eventually get to react native too. or you can pass custom crypto/storage implementations (or contribute!) if you are eager :D

**Comment by addreeh** [comment]
hey, any update with this? I look forward to making an app with it

**Comment by cyan-2048** [comment]
> hey, any update with this? I look forward to making an app with it

judging by what teidesu just said, you can create a fork of @mtcute/web and replace the Wasm crypto with JavaScript/ReactNative equivalents. It is definitely possible.

**Comment by addreeh** [comment]
> > hey, any update with this? I look forward to making an app with it
> 
> judging by what teidesu just said, you can create a fork of @mtcute/web and replace the Wasm crypto with JavaScript/ReactNative equivalents. It is definitely possible.

sadly I don't have the necessary knowledge to do this, I have tried several libraries and none of them have helped me.

**Comment by cyan-2048** [comment]
bro all you have to do is replace these methods with JavaScript libraries, it is not that complicated

https://github.com/mtcute/mtcute/blob/51b392696adbd336d4c97fadc8ceab07412b6890/packages/web/src/crypto.ts

**Comment by cyan-2048** [comment]
let me spoon feed you the libraries you can use:

for gzip/gunzip: https://github.com/101arrowz/fflate
SHA256, SHA1, IGE, CTR: https://github.com/spalt08/cryptography
